lib/spack/spack/test/build_systems.py

```diff
@@ -115,6 +115,27 @@ def test_cmake_std_args(config, mock_packages):
     assert get_std_cmake_args(pkg)
 
 
+def test_cmake_bad_generator(config, mock_packages):
+    s = Spec('cmake-client')
+    s.concretize()
+    pkg = spack.repo.get(s)
+    pkg.generator = 'Yellow Sticky Notes'
+    try:
```

Comment:
@chissg Can you use the `with pytest.raises` syntax (e.g. https://stackoverflow.com/questions/23337471/how-to-properly-assert-that-an-exception-gets-raised-in-pytest)?

Other than that this all looks good